primitives/tests/api.rs

```diff
@@ -0,0 +1,284 @@
+// These derives are the policy of `rust-bitcoin` not Rust API guidelines.
+#[derive(Debug, Clone, PartialEq, Eq)] // All public types implement Debug (C-DEBUG).
+struct Errors {
+    a: transaction::ParseOutPointError,
+    b: relative::IncompatibleHeightError,
+    c: relative::IncompatibleTimeError,
+    d: relative::IncompatibleHeightError,
+    e: relative::IncompatibleTimeError,
+    f: relative::DisabledLockTimeError,
+    g: relative::DisabledLockTimeError,
+    h: script::RedeemScriptSizeError,
+    i: script::WitnessScriptSizeError,
+}
+
+#[test]
+fn api_can_use_units_modules_from_crate_root() {
+    use bitcoin_primitives::{amount, block, fee_rate, locktime, weight};
+}
+
+#[test]
+fn api_can_use_units_types_from_crate_root() {
+    use bitcoin_primitives::{Amount, BlockHeight, BlockInterval, FeeRate, SignedAmount, Weight};
+}
+
+#[test]
+fn api_can_use_all_units_types_from_module_amount() {
+    use bitcoin_primitives::amount::{
+        Amount, Denomination, Display, InputTooLargeError, InvalidCharacterError,
+        MissingDenominationError, MissingDigitsError, OutOfRangeError, ParseAmountError,
+        ParseDenominationError, ParseError, PossiblyConfusingDenominationError, SignedAmount,
+        TooPreciseError, UnknownDenominationError,
+    };
+}
+
+#[test]
+fn api_can_use_modules_from_crate_root() {
+    use bitcoin_primitives::{
+        block, locktime, merkle_tree, pow, script, sequence, transaction, witness,
+    };
+}
+
+#[test]
+fn api_can_use_types_from_crate_root() {
+    use bitcoin_primitives::{
+        Block, BlockHash, BlockHeader, BlockVersion, CompactTarget, OutPoint, Script, ScriptBuf,
+        Sequence, Transaction, TransactionVersion, TxIn, TxMerkleNode, TxOut, Txid, Witness,
+        WitnessCommitment, WitnessMerkleNode, Wtxid,
+    };
+}
+
+#[test]
+fn api_can_use_all_types_from_module_locktime() {
+    use bitcoin_primitives::locktime::relative::{
+        DisabledLockTimeError, IncompatibleHeightError, IncompatibleTimeError, LockTime,
+    };
+    use bitcoin_primitives::locktime::{absolute, relative};
+}
+
+#[test]
+fn api_can_use_all_types_from_module_script() {
+    use bitcoin_primitives::script::{
+        RedeemScriptSizeError, Script, ScriptBuf, ScriptHash, WScriptHash, WitnessScriptSizeError,
+    };
+}
+
+// `Debug` representation is never empty (C-DEBUG-NONEMPTY).
+#[test]
+fn api_all_non_error_types_have_non_empty_debug() {
+    macro_rules! check_debug {
+        ($($t:expr);* $(;)?) => {
+            $(
+                let debug = format!("{:?}", $t);
+                assert!(!debug.is_empty());
+            )*
+        }
+    }
+
+    // All the enums.
+    check_debug! {
+        absolute::LockTime::ZERO;
+        relative::LockTime::ZERO
+    };
+
+    // We abuse `Arbitrary` here to get a quick and dirty instance.
+    let ab: [u8; 32] = [0xab; 32];
+    let mut u = arbitrary::Unstructured::new(&ab);
+    let transaction = Transaction::arbitrary(&mut u).unwrap();
```

Comment:
should this belong in `checked_debug!`?  Interesting idea to abuse Arbitrary like this to create random test data, although is this just testing Arbitrary?  It doesn't appear to be testing that debug is not empty.

Reply:
This variable is used below.